[Bug]: AdenCredentialResponse.from_dict ignores integration_id / integration_type keys

## Describe the Bug

AdenCredentialResponse.from_dict ignores the integration_id and integration_type fields when they are present in the input dictionary.
As a result, credential responses created from normalized data (rather than raw Aden API responses) end up with empty identifiers, causing test failures and incorrect runtime behavior.

## To Reproduce

Run the core test suite with PYTHONPATH=.
```bash
PYTHONPATH=. python -m pytest core -q
```
Observe failures in:
```bash
core/framework/credentials/aden/tests/test_aden_sync.py
```

Inspect AdenCredentialResponse.from_dict and note that it only reads alias or provider, ignoring integration_id passed in the data.

## Expected Behavior

AdenCredentialResponse.from_dict should correctly populate:
- integration_id
- integration_type

by preferring explicitly provided fields in the input dictionary, while still supporting legacy Aden API response formats.

## Screenshots
Not applicable (backend-only change).

## Environment

OS: Windows 11 (also reproducible on Linux/macOS)
Python version: 3.13.x
Docker version: Not applicable

## Configuration

Not applicable, issue reproduces with default test configuration.

## Logs

Relevant test failure output:
```bash
AssertionError: assert '' == 'github'
AssertionError: assert '' == 'hubspot'
```

## Additional Context

This issue affects the correctness of credential handling across the framework.
Tests expect from_dict to work with normalized credential data, not only raw Aden API responses.

The proposed fix preserves backward compatibility by resolving identifiers in a clear priority order:

- Explicit integration_id / integration_type
- Normalized fields
- Legacy Aden API fields (alias, provider)

This aligns behavior with test expectations and improves robustness for future integrations.

--- a/core/framework/credentials/aden/client.py
+++ b/core/framework/credentials/aden/client.py
@@ -139,18 +139,15 @@ class AdenCredentialResponse:
     metadata: dict[str, Any] = field(default_factory=dict)
     """Additional integration-specific metadata."""
 
-
     @classmethod
     def from_dict(
         cls, data: dict[str, Any], integration_id: str | None = None
-    ) -> "AdenCredentialResponse":
+    ) -> AdenCredentialResponse:
         """Create from API response dictionary or normalized credential dict."""
 
         expires_at = None
         if data.get("expires_at"):
-            expires_at = datetime.fromisoformat(
-                data["expires_at"].replace("Z", "+00:00")
-            )
+            expires_at = datetime.fromisoformat(data["expires_at"].replace("Z", "+00:00"))
 
         resolved_integration_id = (
             integration_id
@@ -159,10 +156,7 @@ class AdenCredentialResponse:
             or data.get("provider", "")
         )
 
-        resolved_integration_type = (
-            data.get("integration_type")
-            or data.get("provider", "")
-            )
+        resolved_integration_type = data.get("integration_type") or data.get("provider", "")
         metadata = data.get("metadata")
         if metadata is None and data.get("email"):
             metadata = {"email": data.get("email")}
@@ -178,7 +172,6 @@ class AdenCredentialResponse:
             scopes=data.get("scopes", []),
             metadata=metadata,
         )
-
 
 
 @dataclass

--- a/core/framework/credentials/aden/client.py
+++ b/core/framework/credentials/aden/client.py
@@ -139,24 +139,46 @@ class AdenCredentialResponse:
     metadata: dict[str, Any] = field(default_factory=dict)
     """Additional integration-specific metadata."""
 
+
     @classmethod
     def from_dict(
         cls, data: dict[str, Any], integration_id: str | None = None
-    ) -> AdenCredentialResponse:
-        """Create from API response dictionary."""
+    ) -> "AdenCredentialResponse":
+        """Create from API response dictionary or normalized credential dict."""
+
         expires_at = None
         if data.get("expires_at"):
-            expires_at = datetime.fromisoformat(data["expires_at"].replace("Z", "+00:00"))
+            expires_at = datetime.fromisoformat(
+                data["expires_at"].replace("Z", "+00:00")
+            )
+
+        resolved_integration_id = (
+            integration_id
+            or data.get("integration_id")
+            or data.get("alias")
+            or data.get("provider", "")
+        )
+
+        resolved_integration_type = (
+            data.get("integration_type")
+            or data.get("provider", "")
+            )
+        metadata = data.get("metadata")
+        if metadata is None and data.get("email"):
+            metadata = {"email": data.get("email")}
+        if metadata is None:
+            metadata = {}
 
         return cls(
-            integration_id=integration_id or data.get("alias", data.get("provider", "")),
-            integration_type=data.get("provider", ""),
+            integration_id=resolved_integration_id,
+            integration_type=resolved_integration_type,
             access_token=data["access_token"],
             token_type=data.get("token_type", "Bearer"),
             expires_at=expires_at,
             scopes=data.get("scopes", []),
-            metadata={"email": data.get("email")} if data.get("email") else {},
+            metadata=metadata,
         )
+
 
 
 @dataclass

